core/runtime/binaryen/runtime_external_interface.cpp

```diff
@@ -259,7 +278,69 @@ namespace kagome::runtime::binaryen {
         auto res = extension_->ext_chain_id();
         return wasm::Literal(res);
       }
+
+      /// crypto version 1
+      if (import->base == ext_ed25519_public_keys_v1) {
```

Comment:
Possible makes sense to make hash-map for fast selection. Finding by comparison of string (especially with the same beginning) is too non optimal.